bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/TeslaChannelSelectorProxy.java

```diff
@@ -299,10 +289,11 @@ public State getState(String s, TeslaChannelSelectorProxy proxy, Map<String, Str
             @Override
             public State getState(String s, TeslaChannelSelectorProxy proxy, Map<String, String> properties) {
                 State someState = super.getState(s);
+                BigDecimal value = ((DecimalType) someState).toBigDecimal();
                 if (properties.containsKey("temperatureunits") && properties.get("temperatureunits").equals("F")) {
-                    return super.getState(String.valueOf(celsiusToFahrenheit(((DecimalType) someState))));
+                    return new QuantityType<>(value, ImperialUnits.FAHRENHEIT);
```

Comment:
I think we will need to still convert the the value here from C to F ,  While my  `temperatureunits = "F"` , the returning temperature from tesla is still in C
```json
{
    "battery_heater": false,
    "battery_heater_no_power": false,
    "climate_keeper_mode": "off",
    "defrost_mode": 0,
    "driver_temp_setting": 21.7,
    "fan_status": 0,
    "inside_temp": 26.6,
    "is_auto_conditioning_on": false,
    "is_climate_on": false,
    "is_front_defroster_on": false,
    "is_preconditioning": false,
    "is_rear_defroster_on": false,
    "left_temp_direction": -263,
    "max_avail_temp": 28.0,
    "min_avail_temp": 15.0,
    "outside_temp": 21.5,
    "passenger_temp_setting": 21.7,
    "remote_heater_control_enabled": false,
    "right_temp_direction": -263,
    "seat_heater_left": 0,
    "seat_heater_right": 0,
    "side_mirror_heaters": false,
    "smart_preconditioning": false,
    "timestamp": 1573408265469,
    "wiper_blade_heater": false
}
```

I have not checked if distance values are returned in Km vs Miles, but i will let you know.  We might also need to check if a conversation needs to be done when setting the temp values?

Reply:
Actually, always returning `return new QuantityType<>(value, SIUnits.CELSIUS);` is probably all that needs to be done.

Reply:
And the distance?

Reply:
I get the distance always as miles (although having a european car).
So it seems that Tesla does miles+celsius for everyone. Interesting combination, but well... I'll change the code accordingly. Thanks for testing @digitaldan!

Reply:
fixed.